README.md

```diff
@@ -1 +1,145 @@
-# fe-hometask
\ No newline at end of file
+# React & Rockets
+
+**Description:** Challenge for Adjust Frontend Developer
+
+**Owner:** [adjust](https://github.com/adjust)
+
+**Contacts:** frontend-hiring@adjust.com
+
+---
+
+## Welcome to our little coding exercise! 👋
+
+Here you will be given the opportunity to play with JavaScript and Rockets in the same project. For this, we recommend you to block 2-3 hours of your time to complete it.
+
+**Please make sure you have read this page entirely, before starting the challenge**. If you have any doubts, please reach out to us by [opening an issue](#how-to-request-help) as described at the bottom of this page.
+
+**Importantly**, we would like to give all candidates taking this test the same opportunity to solve the exercise in their own way. Because of this, **please do not fork or share this repo (nor your solution) with anybody 🙏🏻**
+
+<img align="center" src="https://i.imgur.com/ekyJNd9.jpg" width="600">
+
+## Instructions
+
+1. Push your solution to a **private repo** in your **personal Github account**
+2. When you are ready for us to take a look, please add [adjust-frontend-hiring][adjust-frontend-hiring] (GitHub user) as collaborator
+
+## General notes
+
+- The final solution must be a working React application
+- You can use any library you see fit
+- Bringing in best practices such as tests, types, linting will be appreciated
+
+---
+
+## Exercise
+
+### TASK #1 - JavaScript
+
+Implement `prepareData` higher order function, which takes an object of filter params `{year, customerName}`, and returns a function that processes a list of missions, by only showing the ones that were launched in `year` and carried a payload belonging to `customerName`.
+
+**Observations:**
+
+- Missions should appear in inverse chronological order (sort), with the exception of those that carried more payloads should appear first.
+- Payloads are carried in the second stage of a rocket and they can belong to multiple customers.
+- It doesn't matter to which `customerName` 'program' each payload belongs to as long as `customerName` is the customer.
+
+**Example:**
+
+Considering we have the list of missions from this [data fixture][data-fixtures], and the following filter params:
+
+```js
+{
+  year: 2018,
+  customerName: "NASA"
+}
+```
+
+The expected result should be:
```

Comment:
We could also setup a code sandbox with test examples and ask them to copy the tests and make sure the test cases we mentioned are working correctly. So that we could open their solutions and check tests are green?

Reply:
> We could also setup a code sandbox with test examples and ask them to copy the tests and make sure the test cases we mentioned are working correctly. So that we could open their solutions and check tests are green?

As far as I know, we have the following problem: a high rate of passing the home task challenge and a low rate of passing the pair coding interviews.
The idea was to increase the home task complexity.
We can still have tests since the challenge is the same. It's possible to have well-described "contracts" for both tasks and run tests during the home task review.

WDYT? 
Could you elaborate on why you think candidates should be asked to write tests directly? 🙏

Reply:
Mostly for the convenience of us, reviewers. It was convenient to use [elonmusk](https://github.com/adjust/elonmusk) script to check if our base expectations are intact - tests are passing.

It would be nice to still have this convenience – running a test script before reviewing the rest of the submission, so that we know the task is "completed" and we can proceed and evaluate other criteria (code quality, etc).

Reply:
ideally [with the ACs and design specs given], I would write e2e tests that elonmusk would inject to the candidate repo and run them

Reply:
I'm sorry for the late response, had some blockers in the team to resolve
This week I'm preparing a talk for townhall
Can it wait until the end of the week?

Reply:
Hi there, I have opened an adjacent PR, where I tried to summarise some of the points left from the discussion here. Please take a look when you have a chance 🙏 

https://github.com/adjust/fe-hometask/pull/2